doc/api/worker.md

```diff
@@ -0,0 +1,469 @@
+# Worker
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+The `worker` module provides a way to create multiple environments running
+on independent threads, and to create message channels between them. It
+can be accessed using:
+
+```js
+const worker = require('worker');
+```
+
+Workers are useful for performing CPU-intensive JavaScript operations; do not
+use them for I/O, since Node.js’s built-in mechanisms for performing operations
+asynchronously already treat it more efficiently than Worker threads can.
+
+Workers, unlike child processes or when using the `cluster` module, can also
+share memory efficiently by transferring `ArrayBuffer` instances or sharing
+`SharedArrayBuffer` instances between them.
+
+## Example
+
+```js
+const { Worker, isMainThread, postMessage, workerData } = require('worker');
+
+if (isMainThread) {
+  module.exports = async function parseJSAsync(script) {
+    return new Promise((resolve, reject) => {
+      const worker = new Worker(__filename, {
+        workerData: script
+      });
+      worker.on('message', resolve);
+      worker.on('error', reject);
+      worker.on('exit', (code) => {
+        if (code !== 0)
+          reject(new Error(`Worker stopped with exit code ${code}`));
+      });
+    });
+  };
+} else {
+  const { parse } = require('some-js-parsing-library');
```

Comment:
@jasnell As I understand it, that is mostly relevant to a `WebWorker`-style API, where the different set of globals is mostly inspired by the need for access restrictions – which are a non-goal as far as this PR is concerned?

Likewise, I think making `Worker` a global when another thing exists in browsers under the same name is probably not ideal. I’d personally be fine with having it *a* global rather than a builtin module, though.

(There has been some prior work re: building an actual isolated container on top of this. But I don’t think it’s sensible in pursuing that at this point, since Node.js has kind of gotten stuck in a “untrusted code execution is not our job” model – which I find somewhat silly given that that’s the very premise of JS … err, getting offtopic. :smile:)

Reply:
@devsnek No real point except convenience – this might help users avoid extra `.on('workerMessage')` handlers. Dropping it if you think it’s a bad idea is totally fine with me. :)

Reply:
@addaleax it seems like a nice thing to have, i just personally want node to stick as close to web as possible